src/test/java/seedu/address/model/ModelManagerTest.java

```diff
@@ -27,6 +32,43 @@ public void constructor() {
         assertEquals(new UserPrefs(), modelManager.getUserPrefs());
         assertEquals(new GuiSettings(), modelManager.getGuiSettings());
         assertEquals(new AddressBook(), new AddressBook(modelManager.getAddressBook()));
+        assertEquals(FXCollections.observableArrayList(), modelManager.getHistoryCommandList());
+    }
+    @Test
+    public void testGetHistoryCommandList() {
+        Command command = new ClearCommand();
+        String commandText = "clear command";
+
+        modelManager.setCommandHistoryText(commandText);
+        modelManager.addHistoryCommand(command);
+
+        assertEquals(1, modelManager.getHistoryCommandList().size());
+        assertEquals(commandText, modelManager.getHistoryCommandList().get(0).getOriginalCommandText());
+        assertEquals(command, modelManager.getHistoryCommandList().get(0).getCommand());
+    }
+    @Test
+    public void testSetCommandHistoryText() {
+        ModelManager model = new ModelManager();
+        String commandText = "sample command text";
+        modelManager.setCommandHistoryText(commandText);
+        Command command = new ClearCommand();
+        model.addHistoryCommand(command);
+        assertEquals("sample command text", model.getHistoryCommandList().get(0).getOriginalCommandText());
```

Comment:
```suggestion
        assertEquals(commandText, model.getHistoryCommandList().get(0).getOriginalCommandText());
```

might want to use the variable name instead?